libs/api/src/test/java/de/cyface/api/AuthorizationTest.java

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (C) 2020-2022 Cyface GmbH - All Rights Reserved
+ *
+ * This file is part of the Cyface Data Collector.
+ *
+ * The Cyface Data Collector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Cyface Data Collector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface Data Collector. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.api;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.hasSize;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.Validate;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import de.cyface.api.model.Role;
+import de.cyface.api.model.User;
+import de.cyface.apitestutils.TestMongoDatabase;
+import de.cyface.apitestutils.fixture.user.DirectTestUser;
+import de.cyface.apitestutils.fixture.user.TestUser;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.mongo.MongoAuthentication;
+import io.vertx.ext.mongo.MongoClient;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * Tests that authorization of users belonging to different groups works as expected.
+ *
+ * @author Klemens Muthmann
+ * @author Armin Schnabel
+ * @version 1.0.2
+ * @since 6.6.0
+ */
+@ExtendWith(VertxExtension.class)
+public class AuthorizationTest {
```

Comment:
This test of yours now finally moved to `libs/api` from `backend` ... we discussed this before.

Now I resolved the circular dependency as I had to write another integration tests `MeasurementIteratorIntegrationTest` and `MeasurementRetrieverTest`.